api/src/main/java/org/openmrs/validator/AllergyValidator.java

```diff
@@ -82,6 +82,10 @@ public void validate(Object target, Errors errors) {
 			} else if (!allergen.isCoded() && StringUtils.isBlank(allergen.getNonCodedAllergen())) {
 				errors.rejectValue("allergen", "allergyapi.allergen.nonCodedAllergen.required");
 			}
+
+				if (StringUtils.isNumeric(allergy.getAllergen().getNonCodedAllergen())) {
+					errors.rejectValue("allergen", "error.allergyapi.allergy.Allergen.cannotContainNumeric");
+				}
```

Comment:
Remove the extra tab space and configure your IDE using [this](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE)  Also, replace `allergy.getAllergen()` with allergen.